db/design_doc.go

```diff
@@ -86,60 +86,30 @@ func (db *Database) PutDesignDoc(ddocName string, ddoc sgbucket.DesignDoc) (err
 
 const (
 	// viewWrapper_adminViews adds the rev to metadata, and strips the _sync property from the view result
-	viewWrapper_adminViews = `function(doc,meta) {
-	                    var sync = doc._sync;
-	                    if (sync === undefined || meta.id.substring(0,6) == "_sync:")
-	                      return;
-	                    if ((sync.flags & 1) || sync.deleted)
-	                      return;
-	                    delete doc._sync;
-	                    meta.rev = sync.rev;
-						(%s) (doc, meta);
-						doc._sync = sync;}`
-	// viewWrapper_userViews does the same work as viewWrapper_adminViews, and also includes channel information in the view results
-	viewWrapper_userViews = `function(doc,meta) {
-		                    var sync = doc._sync;
-		                    if (sync === undefined || meta.id.substring(0,6) == "_sync:")
-		                      return;
-		                    if ((sync.flags & 1) || sync.deleted)
-		                      return;
-		                    var channels = [];
-		                    var channelMap = sync.channels;
-							if (channelMap) {
-								for (var name in channelMap) {
-									removed = channelMap[name];
-									if (!removed)
-										channels.push(name);
-								}
-							}
-		                    delete doc._sync;
-		                    meta.rev = sync.rev;
-		                    meta.channels = channels;
+	syncViewWrapper = `	function(doc,meta) {
```

Comment:
I don't know if there's enough reused code between adminViews and userViews to try to combine into a single wrapper - keeping them separate makes it easier to parse the differences in functionality at a glance, and avoids the conditional logic inside the map function (which has the overhead of being evaluated on every mutation).